applications/common/backend/src/main/java/pl/ds/kyanite/common/components/models/CodeSnippetComponent.java

```diff
@@ -40,4 +41,12 @@ public class CodeSnippetComponent {
   @ValueMapValue
   @Getter
   private String code;
+
+  @Inject
```

Comment:
Maybe we could use @ValueMapValue? It's a lighter mechanism

Reply:
This could even be achieved using immutable class (final fields) and injection through the constructor:
```java
    private final Long field1;
    private final String field2;
    private final BigDecimal field3;

    @Inject
    public CodeSnippetComponent(@ValueMapValue(name = "field1") Long field1,
                            @ValueMapValue(name = "field2") String field2,
                            @ValueMapValue(name = "field3") BigDecimal field3) {
        this.field1 = field1;
        this.field2 = field2;
        this.field3 = field3;
    }
```

While clear and we have only final fields - one downside of this approach is that the field names must be repeated as a property of ValueMapValue annotation (due that by default, method parameter names are not preserved in compiled java code)

Reply:
From the clean code point of view, it could be the best approach, but from the future maintenance perspective, I'm voting for injecting through the field :(